internal/github/github.go

```diff
@@ -0,0 +1,194 @@
+	var repositoryIds []int64
+	for _, repository := range repositories {
+		repo, _, err := g.client.Repositories.Get(context.Background(), g.orgName, repository)
+		if err != nil {
+			return nil, err
+		}
+		repositoryIds = append(repositoryIds, repo.GetID())
+	}
+	return repositoryIds, nil
+}
+
+// createOrUpdateVariable creates or updates a variable in the organization
+func (g *GitHubWrapper) createOrUpdateVariable(variable *github.ActionsVariable) error {
+	ctx := context.Background()
+	var err error
+	_, res, _ := g.client.Actions.GetOrgVariable(ctx, g.orgName, variable.Name)
+	if res.StatusCode == 404 {
+		_, err = g.client.Actions.CreateOrgVariable(ctx, g.orgName, variable)
+	} else if res.StatusCode == 200 {
+		_, err = g.client.Actions.UpdateOrgVariable(ctx, g.orgName, variable)
+	} else {
+		return errors.New("Unable to get variable, status code: " + strconv.Itoa(res.StatusCode))
+	}
+
+	return err
+}
+
+// getOrgPublicKey retrieves the organization public key
+func (g *GitHubWrapper) getOrgPublicKey() (string, string, error) {
+	ctx := context.Background()
+	publicKey, _, err := g.client.Actions.GetOrgPublicKey(ctx, g.orgName)
+	if err != nil {
+		return "", "", err
+	}
+	return publicKey.GetKey(), publicKey.GetKeyID(), nil
+}
+
+// ecryptSecret encrypts a secret value using the public key
+func encryptSecret(secretValue, publicKey string) (string, error) {
+	publicKeyBytes, err := base64.StdEncoding.DecodeString(publicKey)
+	if err != nil {
+		return "", err
+	}
+
+	var publicKeyArray [32]byte
+	copy(publicKeyArray[:], publicKeyBytes)
+
+	encrypted, err := box.SealAnonymous(nil, []byte(secretValue), &publicKeyArray, nil)
+	if err != nil {
+		return "", err
+	}
+
+	return base64.StdEncoding.EncodeToString(encrypted), nil
+}
+
+// getGitHubAppCredentials retrieves the GitHub App credentials from the environment
+func getGitHubAppCredentials() (int64, int64, []byte, string, error) {
+	appID, exists := os.LookupEnv("GITHUB_APP_ID")
+	if !exists {
+		return 0, 0, nil, "", errors.New("GITHUB_APP_ID environment variable is required")
+	}
+
+	installationID, exists := os.LookupEnv("GITHUB_INSTALLATION_ID")
+	if !exists {
+		return 0, 0, nil, "", errors.New("GITHUB_INSTALLATION_ID environment variable is required")
+	}
+
+	privateKey, exists := os.LookupEnv("GITHUB_PRIVATE_KEY")
+	if !exists {
+		return 0, 0, nil, "", errors.New("GITHUB_PRIVATE_KEY environment variable is required")
+	}
+
+	orgName, exists := os.LookupEnv("GITHUB_ORG_NAME")
+	if !exists {
+		return 0, 0, nil, "", errors.New("GITHUB_ORG_NAME environment variable is required")
+	}
+
+	parsedAppID, err := strconv.ParseInt(appID, 10, 64)
+	if err != nil {
+		return 0, 0, nil, "", errors.New("Failed to parse GITHUB_APP_ID")
+	}
+
+	parsedInstallationID, err := strconv.ParseInt(installationID, 10, 64)
+	if err != nil {
+		return 0, 0, nil, "", errors.New("Failed to parse GITHUB_INSTALLATION_ID")
+	}
```

Comment:
You can also build a function for validating types which returns feedback in the console